lib/Service/OAuthService.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Stripe\Service;
+
+class OAuthService extends \Stripe\Service\AbstractService
+{
+    /**
+     * Sends a request to Stripe's Connect API.
+     *
+     * @param string $method the HTTP method
+     * @param string $path the path of the request
+     * @param array $params the parameters of the request
+     * @param array|\Stripe\Util\RequestOptions $opts the special modifiers of the request
+     *
+     * @return \Stripe\StripeObject the object returned by Stripe's Connect API
+     */
+    protected function requestConnect($method, $path, $params, $opts)
+    {
+        $opts = \Stripe\Util\RequestOptions::parse($opts);
+        if (!isset($opts->apiBase)) {
+            $opts->apiBase = $this->client->getConnectBase();
+        }
+
+        return $this->request($method, $path, $params, $opts);
+    }
+
+    /**
+     * Generates a URL to Stripe's OAuth form.
+     *
+     * @param null|array $params
+     * @param null|array $opts
+     *
+     * @return string the URL to Stripe's OAuth form
+     */
+    public function authorizeUrl($params = null, $opts = null)
+    {
+        $params = $params ?: [];
+
+        $base = ($opts && \array_key_exists('connect_base', $opts)) ? $opts['connect_base'] : $this->client->getConnectBase();
+
+        $params['client_id'] = $this->_getClientId($params);
+        if (!\array_key_exists('response_type', $params)) {
+            $params['response_type'] = 'code';
+        }
+        $query = \Stripe\Util\Util::encodeParameters($params);
+
+        return $base . '/oauth/authorize?' . $query;
+    }
+
+    /**
+     * Use an authoriztion code to connect an account to your platform and
+     * fetch the user's credentials.
+     *
+     * @param null|array $params
+     * @param null|array $opts
+     *
+     * @throws \Stripe\Exception\OAuth\OAuthErrorException if the request fails
+     *
+     * @return StripeObject object containing the response from the API
+     */
+    public function token($params = null, $opts = null)
+    {
+        $params = $params ?: [];
+        $params['client_secret'] = $this->_getClientSecret($params);
+
+        return $this->requestConnect('post', $this->buildPath('/oauth/token'), $params, $opts);
```

Comment:
You don't need to use `buildPath()` since the path is constant.

```suggestion
        return $this->requestConnect('post', '/oauth/token', $params, $opts);
```